core/src/test/java/org/apache/iceberg/TestSchemaAndMappingUpdate.java

```diff
@@ -155,6 +156,42 @@ public void testDeleteColumn() {
     validateUnchanged(mapping, updated);
   }
 
+  @Test
+  public void testModificationWithMetricsMetrics() {
+    NameMapping mapping = MappingUtil.create(table.schema());
+    String mappingJson = NameMappingParser.toJson(mapping);
+
+    table.updateProperties()
+        .set(TableProperties.DEFAULT_NAME_MAPPING, mappingJson)
+        .set("write.metadata.metrics.column.id", "full")
+        .commit();
+
+    AssertHelpers.assertThrows(
+        "Creating metrics for non-existent column fails",
+        ValidationException.class,
+        null,
+        () ->
+          table.updateProperties()
+            .set(TableProperties.DEFAULT_NAME_MAPPING, mappingJson)
+            .set("write.metadata.metrics.column.ids", "full")
+          .commit());
+
+    AssertHelpers.assertThrows(
+        "Deleting a column with metrics fails",
+        ValidationException.class,
+        null,
+        () ->
+          table.updateSchema()
+              .deleteColumn("id")
+          .commit());
+
+    String updatedJson = table.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
+    NameMapping updated = NameMappingParser.fromJson(updatedJson);
+
+    // should not change the mapping
+    validateUnchanged(mapping, updated);
```

Comment:
It seems strange to me that this would be tested in the suite for the schema mapping. This should be independent of mapping. While it is okay to have a test that a mapping is not updated if the validation fails, that should be covered by not updating the schema if metrics property validation fails. Could you start a new suite similar to `TestSchemaAndMappingUpdate` for this instead of adding the tests here?

Reply:
Sure I can make a new suite for this.

Reply:
Sure I'll add a new test suite next.